### ISSUE TITLE
False positives for outermost block of nested structures

### ISSUE THREAD

**Issue by adriaanzon** [tpope/vim-endwise#96]
Given the following vimscript example, where `|` represents the cursor:

```vim
function Foo()|
  if 1
  endif
endfunction
```

when you press `<CR>`, you'll get a false positive.

```vim
function Foo()
  |
endfunction
  if 1
  endif
endfunction
```

This `sh` example also produces a false positive:

```sh
if true; then|
  for item in a b c; do
  done
fi
```

It only seems to happen in filetypes that have an `endwise_addition` that starts with `\=`. When I bring the `endwise_addition` of `vim` down to "end&", it doesn't happen anymore.

**Comment by adriaanzon** [comment]
looks related to #46

**Comment by tpope** [comment]
Just confirmed this is broken in master but works fine in 1.2. If someone could do a bisect and figure out the culprit commit it would help me to figure out a solution.

**Comment by adriaanzon** [comment]
Here you go :) The bug seemed to have been reintroduced several times so I made a little script that tests it on every commit. I have abused the `diff` syntax to show a nice visualisation of where it fails.

For filetype `vim`:

```diff
- d565526 Avoid extraneous block name in endblock (#95)
- 0ed0e73 Workaround to fix syntax group name retrieval issue (#93)
- 0067ced Merge pull request #82 from kassio/master
- e66fb6c Update Elixir syntax group.
- 526fe36 Merge pull request #81 from glittershark/haskell-cpp
- 7b03a17 Enable CPP endwise-ing in Haskell
- f06abe3 Merge pull request #79 from sukima/feature/disable-in-command-line-window
- d31c924 Fix command-line-window
- 4fab89d Merge pull request #78 from gabesoft/ultisnips-and-augroup
- 6287c77 Added endwise functionality for augroup
+ 0791920 Added support for ultisnips
+ 1fa14da Merge pull request #77 from kevinushey/feature/endwise-functions
+ 4b57ccb expose 'crend()' in global 'Endwise*' functions
+ c72da70 Merge pull request #64 from blueyed/add-htmldjango
+ bba43b8 Merge pull request #68 from padde/master
+ 568d918 Merge pull request #73 from yous/support-crystal
+ b97bb84 Support Crystal (Fix #63)
+ 2b25409 Fix elixir support for inline do: keywords
+ 96eb117 Merge pull request #67 from danieljaouen/master
+ 2e2b4c2 Improve Elixir support
+ 6b83716 Merge pull request #66 from danieljaouen/master
+ 9addc41 Add Elixir support
+ 6c2d288 Remove broken Elixer support
+ 173b2f1 Add support for ft=htmldjango
+ ab5f7f9 Revert "Add `do` to the list of Elixir keywords"
+ 4f563fd Merge pull request #60 from whatyouhide/patch-1
+ 8ae7506 Add `do` to the list of Elixir keywords
+ 38541f3 Experimental abbreviations
+ ea2a7f2 endwise.vim 1.2
+ fe4c49e Fix conflict with unmentionable
+ dc42ba6 Special case \= additions
- cdde2dd Add support for Ruby def modifiers
- c381da3 Try a different header style
- c0a825f Remove more cargo culting
- 34dbe76 Use then not if to trigger fi for sh
- 254861f Limit b:endwise_words to actual words
- 6eee950 Fix C match detection
- 7225610 Fix duplicates on <C-X><CR>
- ef7356c Extract contribution guidelines
- 8262382 Use # style markdown headings
- c3d8fe0 Remove unnecessary keyword
- aa89cde Merge pull request #54 from Keithbsmiley/master
- 51e90c1 Add Objective-C endwise rules
- 228f9ff Merge pull request #52 from telemachus/lua-local-funcs
- 31f8fa0 Add support for local functions in Lua
- 1e651aa Merge pull request #47 from blueyed/remap-plug-cr
- e320dc1 Mapping: keep/forward any <Plug>\w+CR mappings
- a12bad2 Merge pull request #45 from danieljaouen/master
- abf0072 Add 'fn' keyword to Elixir words and pattern.
- b2a7fd5 Merge pull request #44 from lazywei/master
- 3360f9d Change matlab syn_group.
- 71eb35e Merge pull request #43 from lazywei/master
- 63a538e Add matlab file support.
- 71a75f7 Angrily remove someone else's whitespace
- 9346e24 Merge pull request #36 from richsoni/add_elixir_support
- eb07ab7 Updated endwise_pattern to match elixir endwise_words
- 02abde6 Added Support For Elixir (via elixirKeyword)
- 78df3d5 Remove silly map
- 9f05e4c ruby: assignments to endwise pattern don't break completion
- fd869a7 endwise.vim 1.1
- a45d4a6 Fix detection of #endif
- 498f566 Add xdefaults to preprocessor support
- 11be5c9 Add C and C++ PreProcessor support
- f664ebd Fix remap issue
- 33f9ee0 Add `g:endwise_no_mappings` to allow skipping maps
- 2a61d25 Use "SNR" instead of "SID" in maparg() comparison
- 74306aa README: Fix link to script page in vim.org
- 9a9d199 Add a README
- 39124dd Add license
- b6f95cc Fix Bourne shell support
+ 6e55ac1 Remove debugging statement
+ 84960ea Switch to 2 space indent
+ 8adfe1d Clean up boilerplate
+ 335f223 Sort
+ 86e1d4a Merge pull request #13 from camthompson/master
+ 10c312d Detect delimitMate mapping and append accordingly
+ d8c887b Add support for sh syntax
+ b674cb5 Add support for Lua
```

For the `sh` filetype, the bug seems to have been around ever since support for `sh` was added:

```diff
- d565526 Avoid extraneous block name in endblock (#95)
- 0ed0e73 Workaround to fix syntax group name retrieval issue (#93)
- 0067ced Merge pull request #82 from kassio/master
- e66fb6c Update Elixir syntax group.
- 526fe36 Merge pull request #81 from glittershark/haskell-cpp
- 7b03a17 Enable CPP endwise-ing in Haskell
- f06abe3 Merge pull request #79 from sukima/feature/disable-in-command-line-window
- d31c924 Fix command-line-window
- 4fab89d Merge pull request #78 from gabesoft/ultisnips-and-augroup
- 6287c77 Added endwise functionality for augroup
- 0791920 Added support for ultisnips
- 1fa14da Merge pull request #77 from kevinushey/feature/endwise-functions
- 4b57ccb expose 'crend()' in global 'Endwise*' functions
- c72da70 Merge pull request #64 from blueyed/add-htmldjango
- bba43b8 Merge pull request #68 from padde/master
- 568d918 Merge pull request #73 from yous/support-crystal
- b97bb84 Support Crystal (Fix #63)
- 2b25409 Fix elixir support for inline do: keywords
- 96eb117 Merge pull request #67 from danieljaouen/master
- 2e2b4c2 Improve Elixir support
- 6b83716 Merge pull request #66 from danieljaouen/master
- 9addc41 Add Elixir support
- 6c2d288 Remove broken Elixer support
- 173b2f1 Add support for ft=htmldjango
- ab5f7f9 Revert "Add `do` to the list of Elixir keywords"
- 4f563fd Merge pull request #60 from whatyouhide/patch-1
- 8ae7506 Add `do` to the list of Elixir keywords
- 38541f3 Experimental abbreviations
- ea2a7f2 endwise.vim 1.2
- fe4c49e Fix conflict with unmentionable
- dc42ba6 Special case \= additions
- cdde2dd Add support for Ruby def modifiers
- c381da3 Try a different header style
- c0a825f Remove more cargo culting
- 34dbe76 Use then not if to trigger fi for sh
- 254861f Limit b:endwise_words to actual words
- 6eee950 Fix C match detection
- 7225610 Fix duplicates on <C-X><CR>
- ef7356c Extract contribution guidelines
- 8262382 Use # style markdown headings
- c3d8fe0 Remove unnecessary keyword
- aa89cde Merge pull request #54 from Keithbsmiley/master
- 51e90c1 Add Objective-C endwise rules
- 228f9ff Merge pull request #52 from telemachus/lua-local-funcs
- 31f8fa0 Add support for local functions in Lua
- 1e651aa Merge pull request #47 from blueyed/remap-plug-cr
- e320dc1 Mapping: keep/forward any <Plug>\w+CR mappings
- a12bad2 Merge pull request #45 from danieljaouen/master
- abf0072 Add 'fn' keyword to Elixir words and pattern.
- b2a7fd5 Merge pull request #44 from lazywei/master
- 3360f9d Change matlab syn_group.
- 71eb35e Merge pull request #43 from lazywei/master
- 63a538e Add matlab file support.
- 71a75f7 Angrily remove someone else's whitespace
- 9346e24 Merge pull request #36 from richsoni/add_elixir_support
- eb07ab7 Updated endwise_pattern to match elixir endwise_words
- 02abde6 Added Support For Elixir (via elixirKeyword)
- 78df3d5 Remove silly map
- 9f05e4c ruby: assignments to endwise pattern don't break completion
- fd869a7 endwise.vim 1.1
- a45d4a6 Fix detection of #endif
- 498f566 Add xdefaults to preprocessor support
- 11be5c9 Add C and C++ PreProcessor support
- f664ebd Fix remap issue
- 33f9ee0 Add `g:endwise_no_mappings` to allow skipping maps
- 2a61d25 Use "SNR" instead of "SID" in maparg() comparison
- 74306aa README: Fix link to script page in vim.org
- 9a9d199 Add a README
- 39124dd Add license
- b6f95cc Fix Bourne shell support
- 6e55ac1 Remove debugging statement
- 84960ea Switch to 2 space indent
- 8adfe1d Clean up boilerplate
- 335f223 Sort
- 86e1d4a Merge pull request #13 from camthompson/master
- 10c312d Detect delimitMate mapping and append accordingly
- d8c887b Add support for sh syntax
+ b674cb5 Add support for Lua
```
